plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1996,7 +2000,7 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
         DeltaLakeTableHandle handle = (DeltaLakeTableHandle) tableHandle;
         Table table = metastore.getTable(handle.getSchemaName(), handle.getTableName())
                 .orElseThrow(() -> new TableNotFoundException(handle.getSchemaTableName()));
-        if (table.getTableType().equals(MANAGED_TABLE.name())) {
+        if (table.getTableType().equals(MANAGED_TABLE.name()) && !metastoreType.equalsIgnoreCase("glue")) {
```

Comment:
I feel it would be better to leave a comment why only glue is allowed. 

Also, how about allowing file metastore for local testing purpose in a separated commit/PR?